argparse_test.go

```diff
@@ -2746,3 +2746,215 @@ func TestCommandHelpSetSnameOnly(t *testing.T) {
+	parser := NewParser("posint", "")
+	intval := parser.Int("i", "integer", &Options{Positional: true})
+	strval := parser.String("s", "string", &Options{Positional: true})
+	floatval := parser.Float("f", "floats", &Options{Positional: true})
+
+	if err := parser.Parse(testArgs1); err != nil {
+		t.Error(err.Error())
+	} else if *intval != 5 {
+		t.Error("Intval did not match expected")
+	} else if *strval != "abc" {
+		t.Error("Strval did not match expected")
+	} else if *floatval != 1.0 {
+		t.Error("Floatval did not match expected")
+	}
+}
+
+func TestCommandPositionalsErr(t *testing.T) {
+	errArgs1 := []string{"posint", "abc", "abc", "1.0"}
+	parser := NewParser("posint", "")
+	_ = parser.Int("i", "cool", &Options{Positional: true})
+	_ = parser.String("s", "string", &Options{Positional: true})
+	_ = parser.Float("f", "floats", &Options{Positional: true})
+
+	if err := parser.Parse(errArgs1); err == nil {
+		t.Error("String argument accepted for integer")
+	} else if err.Error() != "[cool] bad integer value [abc]" {
+		t.Error(err.Error())
+	}
+}
+
+func TestPos1(t *testing.T) {
+	testArgs1 := []string{"pos", "subcommand1", "-i=2", "abc"}
+	parser := NewParser("pos", "")
+
+	strval := parser.String("s", "string", &Options{Positional: true})
+	com1 := parser.NewCommand("subcommand1", "beep")
+	intval := com1.Int("i", "integer", nil)
+
+	if err := parser.Parse(testArgs1); err != nil {
+		t.Error(err.Error())
+	} else if *strval != "abc" {
+		t.Error("Strval did not match expected")
+	} else if *intval != 2 {
+		t.Error("intval did not match expected")
+	}
+}
+
+func TestPos2(t *testing.T) {
+	testArgs1 := []string{"pos", "subcommand1", "a123"}
+	parser := NewParser("pos", "")
+
+	strval := parser.String("s", "string", &Options{Positional: true})
+	com1 := parser.NewCommand("subcommand1", "beep")
+	intval := com1.Int("i", "integer", nil)
+
+	if err := parser.Parse(testArgs1); err != nil {
+		t.Error(err.Error())
+	} else if *strval != "a123" {
+		t.Error("Strval did not match expected")
+	} else if *intval != 0 {
+		t.Error("intval did not match expected")
+	}
+}
+
+func TestPos3(t *testing.T) {
+	testArgs1 := []string{"pos", "subcommand1", "xyz", "--integer", "3"}
+	parser := NewParser("pos", "")
+
+	strval := parser.String("s", "string", &Options{Positional: true})
+	com1 := parser.NewCommand("subcommand1", "beep")
+	intval := com1.Int("i", "integer", nil)
+
+	if err := parser.Parse(testArgs1); err != nil {
+		t.Error(err.Error())
+	} else if *strval != "xyz" {
+		t.Error("Strval did not match expected")
+	} else if *intval != 3 {
+		t.Error("intval did not match expected")
+	}
+}
+
+func TestPos4(t *testing.T) {
+	testArgs1 := []string{"pos", "abc"}
+	parser := NewParser("pos", "")
+
+	strval := parser.String("s", "string", &Options{Positional: true})
+	com1 := parser.NewCommand("subcommand1", "beep")
+	intval := com1.Int("i", "integer", nil)
+
+	if err := parser.Parse(testArgs1); err != nil {
+		t.Error(err.Error())
```

Comment:
This test will fail with current setup. Parser requires `subcommand1`

Reply:
Fixed up the test to match this behavior after pulling latest from master.